### PR TITLE
Rename sql alchemy

### PR DESCRIPTION
**Pull request by wild-endeavor**
Signed-off-by: wild-endeavor <wild-endeavor@users.noreply.github.com>

# TL;DR
Ketan brought up a good point.  We shouldn't use folders that might interfere with python importing logic (i.e. names that have a `.` in them) for these top level microlib folders.

These folders btw don't really matter too much.  The name of the pypi package is determined by the `setup.py` file.  However, if you `cd plugins && pip install -e .` (as part of developer setup for instance) and then you run `import sqlalchemy`, if the top level folder is named `sqlalchemy` then you get the plugin folder instead of the actual sqlalchemy library.  So we just make the name something that users won't really ever import.

In case you can't see the full path in the diff...
```
renamed:    plugins/flytekit.sqlalchemy/flytekitplugins/sqlalchemy/__init__.py -> plugins/flytekitplugins-sqlalchemy/flytekitplugins/sqlalchemy/__init__.py
renamed:    plugins/flytekit.sqlalchemy/flytekitplugins/sqlalchemy/task.py -> plugins/flytekitplugins-sqlalchemy/flytekitplugins/sqlalchemy/task.py
renamed:    plugins/flytekit.sqlalchemy/setup.py -> plugins/flytekitplugins-sqlalchemy/setup.py
```

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

### CODE REVIEW

**Comment by kumare3** [comment]
I think hyphens are not good for python packages?

**Review by cosmicBboy — APPROVED** [review]

File: plugins/setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "flytekitplugins-awssagemaker": "awssagemaker",
     "flytekitplugins-kftensorflow": "kftensorflow",
     "flytekitplugins-pandera": "pandera",
-    "flytekitplugins-sqlalchemy": "flytekit.sqlalchemy",
+    "flytekitplugins-sqlalchemy": "flytekitplugins-sqlalchemy",
     "flytekitplugins-dolt": "flytekitplugins.dolt",
```
[the remark this answers is not in the document]
Comment (wild-endeavor) [reply plugins/setup.py:19]: we should, but I'll let @max-hoffman do that in one of his PRs.
